### PR TITLE
Feature/category button

### PR DESCRIPTION
**Pull request by Neeze**
# Category Menu Button

I re-design the category menu button and split it into discrete reusable components.
Main button
Item button (Item button has default size and small Item button is used in sub menu)
Category Board is where submenus are rendered.
Demo on playground below!

All components work well on medium and large screen sizes. In small screen size, it is said that we must create new components for mobile.

# Checklist

- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] Already rebased against main branch.

# Screenshots
demo behaviour
https://github.com/its-rav/namvietstore-fe/assets/60569049/e8a7fccc-0d20-4a1e-aa6e-546436e9b420

Screenshot 2023-10-03 154231
Screenshot 2023-10-03 154243
Screenshot 2023-10-03 154314
Screenshot 2023-10-03 154218

Provide screenshots or videos of the changes made if any.

### CODE REVIEW

File: playground/src/CategoryMenuButton.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
+import type { Story, StoryDefault } from "@ladle/react";
+import React from "react";
+import "regenerator-runtime/runtime";
+import "@namviet-fe/core-ui/dist/style.css";
+
+import {type CategoryMenu, CategoryUsage } from "@namviet-fe/core-ui";
+import { StripsIcon } from "@namviet-fe/core-ui";
+import { KeyIcon } from "@namviet-fe/core-ui";
+import { CategoryButton, type CategoryMenuButtonProps} from "@namviet-fe/core-ui";
+import {type ItemProps, CategoryItem} from "@namviet-fe/core-ui"
```
Comment (phongduong-dp) [inline playground/src/CategoryMenuButton.stories.tsx:10]: @Neeze delete unused import

Comment (Neeze) [reply playground/src/CategoryMenuButton.stories.tsx:10]: I solved it

**Comment by its-rav** [comment]
title size seems off to me. Too big

**Comment by its-rav** [comment]
button missing border-radius !? This is why I want to reuse the Button component

File: packages/core-components/src/components/FeaturesCard.tsx
Comment (its-rav) [inline packages/core-components/src/components/FeaturesCard.tsx:1]: Why do we have changes related to Feature/Service Card

File: packages/core-components/src/components/category/CategoryProps.tsx
Comment (its-rav) [inline packages/core-components/src/components/category/CategoryProps.tsx:1]: we don't have any jsx here so I guess .ts would be a better extension for this?

File: packages/core-components/src/components/category/CategoryUsage.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import CategoryBoard from '@/components/category/CategoryBoard';
+import CategoryButton from '@/components/category/CategoryButton';
+import CategoryItem from '@/components/category/CategoryItem';
+import {
+  CategoryMenu,
+  CategorySubMenu,
+} from '@/components/category/CategoryProps';
+
+const CategoryUsage: React.FC<CategoryMenu> = ({
```
Comment (its-rav) [inline packages/core-components/src/components/category/CategoryUsage.tsx:11]: lol, CategoryMenu/CategoryMenuComponent should be fine

File: packages/core-components/src/components/category/CategoryItem.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  handleMouseEnter?: () => void;
+  handleMouseLeave?: () => void;
+};
+
+const CategoryItem: React.FC<ItemProps> = ({
+  icon,
+  link,
+  children,
+  boldTitle,
+  small,
```
Comment (its-rav) [inline packages/core-components/src/components/category/CategoryItem.tsx:19]: what is this `small` prop. Not sure how can I understand this

**Review by its-rav — COMMENTED** [review]
How can we navigate on clicking to any of the items!?

**Review by its-rav — CHANGES_REQUESTED** [review]
Please double check your components, make sure others can understand and use it ?

File: packages/core-components/src/components/FeaturesCard.tsx
Comment (its-rav) [reply packages/core-components/src/components/FeaturesCard.tsx:1]: Why do you resolve this !? the change is still here !?

File: packages/core-components/src/components/category/CategoryItem.tsx (hunk 3 above)
Comment (its-rav) [reply packages/core-components/src/components/category/CategoryItem.tsx:19]: same question. Have you fixed this!? why do you mark this as resolved?!

File: packages/core-components/src/components/category/CategoryItem.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        <div
+          className={
+            'flex items-center justify-left gap-2.5 bg-white text-white lg:w-56 lg:py-2.5 md:ps-5 md:w-52 md:py-2.5 ' +
+            'mt-1 hover:bg-secondary-50 transition ease-in-out hover:scale-105 hover:font-bold hover:shadow-md'
+          }
+        >
+          <div className={'lg:w-5 md:w-4'}>{icon}</div>
+          <a
+            href={link}
+            className={`text-black md:text-sm ${boldTitle ? 'font-bold' : ''}`}
```
Comment (its-rav) [inline packages/core-components/src/components/category/CategoryItem.tsx:41]: I recall mentioning this earlier. Isn't isTitleBold better!?

File: playground/src/ServiceCard.stories.tsx [hunk 5]
```diff
@@ -6,20 +6,20 @@ import "regenerator-runtime/runtime";
 import "@namviet-fe/core-ui/dist/style.css";
```
Comment (its-rav) [inline playground/src/ServiceCard.stories.tsx:6]: same here. Unrelated changes

File: playground/.ladle/config.mjs [hunk 6]
```diff
@@ -4,7 +4,7 @@ export default {
   addons: {
     width: {
       options: {
-        xsmall: 375,
+        xsmall: 320,
```
Comment (its-rav) [inline playground/.ladle/config.mjs:7]: why do we have this change!? do we have any discussion on why should we have xsmall 320!?

**Review by its-rav — CHANGES_REQUESTED** [review]

File: packages/core-components/src/components/category/CategoryButton.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
```
Comment (its-rav) [inline packages/core-components/src/components/category/CategoryButton.tsx:1]: this one good 🧠

File: packages/core-components/src/components/category/CategoryItem.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+export type ItemProps = {
+  icon?: React.ReactNode;
+  link?: string;
+  children: string;
```
Comment (its-rav) [inline packages/core-components/src/components/category/CategoryItem.tsx:6]: children as string !? for `children` you can just do this
```
type ItemProps = {
    // ...props
} & PropsWithChildren;
```

File: packages/core-components/src/components/category/CategoryItem.tsx (hunk 3 above)
Comment (its-rav) [reply packages/core-components/src/components/category/CategoryItem.tsx:19]: I get what is `small` prop now. But still, why do we have to have this condition? Create another component and call it a day

**Comment by its-rav** [comment]
close cause of inactivity

**Pull request closed without merging**